cdap-app-templates/cdap-etl/cdap-etl-api-spark/src/main/java/io/cdap/cdap/etl/api/batch/SparkApiPreference.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api.batch;
+
+/**
+ * List of Spark API modes that plugin prefers to operate in.
+ *
+ * @see SparkCompute#getApiPreference
+ */
+public enum SparkApiPreference {
+  /**
+   * Default mode that indicates that only RDD mode is supported.
+   */
+  REQUIRES_RDD,
+  /**
+   * Mode that indicates that both modes are supported with no preference.
+   */
+  SUPPORTS_DATAFRAME,
+  /**
+   * Mode that indicates that DataFrame mode is preferred. Platform may still use RDD mode if
```

Comment:
Agree. Let me add an interface. Anyway, that's how we usually do it, so that's a right way

Reply:
The only problem with this is that we don't have preference anymore and DynamicSpark plugin would need to convert to RDD depending on underlying implementation. I wanted to take this burden from plugin and allow plugin to tell what it wants dynamically (e.g. for DynamicSpark this would depend on user code)

Reply:
Another option is to give plugin some kind of object where is can choose dynamically from the variety of options

Reply:
Probably let me take it out from this PR, we can discuss it in the follow-up